sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -122,6 +131,8 @@ private[sbtplugin] object ScalaJSPluginInternal {
         box.ensure(linkerImpl.clearableLinker(config))
       },
 
+      scalaJSLinker in legacyKey := (scalaJSLinker in key).value,
```

Comment:
This is not right. It will silently break compatibility in ways that are hard to debug. Although it's a bit of a specialized use case, if a user had specified
```scala
scalaJSLinker in (Compile, fastOptJS) := something
```
that setting will now be ignored. IMO we should do the reverse. Set `scalaJSLinker in legacyKey` first, then derive `scalaJSLinker in key := (scalaJSLinker in legacyKey).value`.

Reply:
Writing to `scalaJSLinker` is specified to be an unstable API, but fair enough, for consistency with everything else, we might as well do it.